### PR TITLE
allocateNetwork: fix network sandbox not cleaned up on failure

### PR DESCRIPTION
**Pull request by thaJeztah**
The defer function was checking for the local `err` variable, not on the error that was returned by the function. As a result, the sandbox would never be cleaned up for containers that used "none" networking, and a failiure occured during setup.

The second commit is a small refactor; allocateNetwork() can return early, in which case these variables were unused, so assign them later in the function.

Looks like this was introduced in https://github.com/moby/moby/pull/31996 (which was created to address https://github.com/moby/moby/issues/31597)

**- Description for the changelog**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@arkodg @xinfengliu  @AkihiroSuda PTAL

File: daemon/container_operations.go [hunk 1]
```diff
@@ -594,7 +596,7 @@ func (daemon *Daemon) allocateNetwork(container *container.Container) error {
 			}
 			updateSandboxNetworkSettings(container, sb)
 			defer func() {
```
Comment (xinfengliu) [inline daemon/container_operations.go:598]: The purpose of `defer` here is to handle the err in following `container.WriteHostConfig()`. You are right @thaJeztah .

**Review by xinfengliu — COMMENTED** [review]
Looks good to me.

File: daemon/container_operations.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/container_operations.go:598]: Yes, it's very easy to overlook the problem. My eye just fell on it, and I thought: "that doesn't look right!"

**Review by xinfengliu — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
LGTM

**Review by tiborvass — APPROVED** [review]

**Pull request merged**